src/main/scala/io/iohk/ethereum/blockchain/sync/regular/BlockFetcherState.scala

```diff
@@ -188,11 +220,31 @@ object BlockFetcherState {
     importer = importer,
     readyBlocks = Queue(),
     waitingHeaders = Queue(),
-    isFetchingHeaders = false,
-    isFetchingBodies = false,
+    fetchingHeadersState = NoHeadersFetched,
+    fetchingBodiesState = NoBodiesFetched,
     stateNodeFetcher = None,
     lastBlock = lastBlock,
     knownTop = lastBlock + 1,
     blockProviders = Map()
   )
+
+  trait FetchingHeadersState
+  case object NoHeadersFetched extends FetchingHeadersState
+  case object AwaitingHeaders extends FetchingHeadersState
```

Comment:
what about tracking parameters for request made? It would allow to track (and at least log for now) case which Konrad mentioned - received response in a moment when we're waiting for different one

Reply:
I'd do this on a separate task, one that adds the BlockChainDataFetcher class and all the validations of requests from there, wdyt?

(also for now I don't think it will be that straightforward integrating the BlockChainDataFetcher with the current actors that we have)